fieldservice/models/fsm_location.py

```diff
@@ -344,6 +344,16 @@ def _check_location_recursion(self):
             raise ValidationError(_('You cannot create recursive location.'))
         return True
 
+    @api.onchange('country_id')
+    def _onchange_country_id(self):
+        if self.country_id and self.country_id != self.state_id.country_id:
+            self.state_id = False
+
+    @api.onchange('state_id')
+    def _onchange_state(self):
+        if self.state_id.country_id:
+            self.country_id = self.state_id.country_id
```

Comment:
```
    @api.onchange('state_id')
    def _onchange_state(self):
        self.partner_id._onchange_state()
```

Reply:
self.partner_id._onchange_state() and self.partner_id._onchange_country_id() did not work for either field when I was working on these. Tested again to be doubly sure and neither method is called.